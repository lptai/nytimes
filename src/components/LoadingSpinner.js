import React from 'react';
import styled, { keyframes } from 'styled-components';
import { Box } from 'grid-styled';

const upAndDown = keyframes`
  0% {
    opacity: 0;
    transform: translateY(0);
  }
  25% {
    opacity: 1;
    transform: translateY(-10px);
  }
  75% {
    opacity: 1;
    transform: translateY(-10px);
  }
  100% {
    opacity: 0;
    transform: translateY(0);
  }
`;

const Spinner = styled.svg.attrs({
  className: 'spinner',
})`
  display: block;
  margin: 0 auto;
  fill: #000;
  & > circle {
    animation-name: ${upAndDown};
    animation-duration: 2s;
    animation-timing-function: cubic-bezier(0.05, 0.2, 0.35, 1);
    animation-iteration-count: infinite;

    &:nth-child(2) {
      animation-delay: 0.18s;
    }
    &:nth-child(3) {
      animation-delay: 0.36s;
    }
  }
`;

const LoadingSpinner = () => (
  <Box py={[2, 4]}>
    <Spinner width="60" height="20" viewBox="0 0 60 20" xmlns="http://www.w3.org/2000/svg">
      <circle cx="7" cy="15" r="4" />
      <circle cx="30" cy="15" r="4" />
      <circle cx="53" cy="15" r="4" />
    </Spinner>
  </Box>
);

export default LoadingSpinner;
