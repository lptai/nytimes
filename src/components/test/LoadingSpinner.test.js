import React from 'react';

import 'jest-styled-components';

import TestRenderer from 'react-test-renderer';
import LoadingSpinner from './../LoadingSpinner';

test('Loading spinner', () => {
  const testRenderer = TestRenderer.create(<LoadingSpinner />);
  expect(testRenderer).toMatchSnapshot();
});
