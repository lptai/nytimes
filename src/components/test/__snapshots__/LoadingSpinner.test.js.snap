// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading spinner 1`] = `
.c0 {
  box-sizing: border-box;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c1 {
  display: block;
  margin: 0 auto;
  fill: #000;
}

.c1 > circle {
  -webkit-animation-name: kLZGqs;
  animation-name: kLZGqs;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: cubic-bezier(0.05,0.2,0.35,1);
  animation-timing-function: cubic-bezier(0.05,0.2,0.35,1);
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

.c1 > circle:nth-child(2) {
  -webkit-animation-delay: 0.18s;
  animation-delay: 0.18s;
}

.c1 > circle:nth-child(3) {
  -webkit-animation-delay: 0.36s;
  animation-delay: 0.36s;
}

@media screen and (min-width:40em) {
  .c0 {
    padding-top: 32px;
    padding-bottom: 32px;
  }
}

<div
  className="c0"
>
  <svg
    className="spinner c1"
    height="20"
    viewBox="0 0 60 20"
    width="60"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="7"
      cy="15"
      r="4"
    />
    <circle
      cx="30"
      cy="15"
      r="4"
    />
    <circle
      cx="53"
      cy="15"
      r="4"
    />
  </svg>
</div>
`;
